loopy/target/c/__init__.py

```diff
@@ -1185,4 +1218,105 @@ def get_host_ast_builder(self):
 
 # }}}
 
+# {{{ c target with vector extension
+
+
+class DTypeRegistryWrapperVec(DTypeRegistryWrapper):
```

Comment:
Are `DTypeRegistryWrapperVec` and `CVecASTBuilder(CASTBuilder)` and `CVecTarget` not going to be specialised anymore? As in we would go through `DTypeRegistryWrapper` and `CASTBuilder` and `CTarget` and special case there? Or do we keep these as is?